### ISSUE TITLE
arc & threads: doesn't pass the reference forwarding test

### ISSUE THREAD

**Issue by cooldome** [nim-lang/Nim#13805]
The object is not destroyed:
```nim
import threadpool, os
type
  MyObj = object
    p: int

proc `=destroy`(x: var MyObj) = 
  if x.p != 0:
    echo "destroyed"

proc thread4(x: ref MyObj): ref MyObj =
  os.sleep(1000)
  result = x

proc ref_forwarding_test = 
  var x = new(MyObj)
  x[].p = 2
  var y = spawn thread4(x)
  sync()

ref_forwarding_test()
```

Destructor is not invoked.

**Comment by cooldome** [comment]
I believe the right fix is to pass thread wrapper procs through transf and inject destructors  as you need control flow graph to figure out the correct arguments passing scheme. 

Update:  transf and inject destructors were not required in the end

**Comment by cooldome** [comment]
Added fix for #13805

Along the way I have to fix an issue that for
```nim
proc test(x: ref MyObj)
```
hasDestructor(x.typ) returned false simply because `createTypeBoundOps` was invoked for it.
`createTypeBoundOps` is also needed for proc's return types
